### PR TITLE
feat (team service account)  Manage Team Based Service Account as Terraform resource

### PR DESCRIPTION
**Pull request by Shadow649**
This PR adds support for managing team service account via `sysdig_team_service_account` resource for both monitor and secure teams.

### CODE REVIEW

File: sysdig/internal/client/v2/team_service_account.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var TeamServiceAccountNotFound = errors.New("group mapping not found")
```
Comment (filiptubic) [inline sysdig/internal/client/v2/team_service_account.go:10]: Copy pasted error message from group mapping

File: sysdig/resource_sysdig_team_service_account.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	if err != nil {
+		diag.FromErr(err)
+	}
+
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		diag.FromErr(err)
+	}
+
+	teamServiceAccount, err := client.GetTeamServiceAccountById(ctx, id)
```
Comment (filiptubic) [inline sysdig/resource_sysdig_team_service_account.go:75]: Rename to `GetTeamServiceAccountByID`

File: sysdig/resource_sysdig_team_service_account_test.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
+//go:build tf_acc_sysdig_monitor || tf_acc_sysdig_secure
```
Comment (filiptubic) [inline sysdig/resource_sysdig_team_service_account_test.go:1]: Is the feature available for IBM?

File: sysdig/resource_sysdig_team_service_account.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
```
[the remark this answers is not in the document]
Comment (Shadow649) [reply sysdig/resource_sysdig_team_service_account.go]: replaced all string with constants, thanks for the suggestion

File: sysdig/resource_sysdig_team_service_account.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func teamServiceAccountFromResourceData(d *schema.ResourceData) *v2.TeamServiceAccount {
+	return &v2.TeamServiceAccount{
+		Name:           d.Get("name").(string),
```
Comment (filiptubic) [inline sysdig/resource_sysdig_team_service_account.go:158]: Change here as well

File: sysdig/resource_sysdig_team_service_account.go [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+		TeamRole:       d.Get("role").(string),
+		ExpirationDate: int64(d.Get("expiration_date").(int) * 1000),
+		TeamId:         d.Get("team_id").(int),
+		SystemRole:     d.Get("system_role").(string),
+		ApiKey:         d.Get("api_key").(string),
+	}
+}
+
+func teamServiceAccountToResourceData(teamServiceAccount *v2.TeamServiceAccount, d *schema.ResourceData) error {
+	err := d.Set("name", teamServiceAccount.Name)
```
Comment (filiptubic) [inline sysdig/resource_sysdig_team_service_account.go:168]: Also here

File: sysdig/resource_sysdig_team_service_account_test.go (hunk 3 above)
Comment (Shadow649) [reply sysdig/resource_sysdig_team_service_account_test.go:1]: For now we don't support team service account in IBM envs.

**Review by filiptubic — APPROVED** [review]
🚀

**Pull request merged**